generic3g/MAPL_Generic.F90

```diff
@@ -217,6 +218,7 @@ subroutine gridcomp_get(gridcomp, unusable, &
       type(OuterMetaComponent), pointer, optional, intent(out) :: outer_meta
       class(Logger_t), optional, pointer, intent(out) :: logger
       type(HierarchicalRegistry), optional, pointer, intent(out) :: registry
+      type(ESMF_Geom), optional, intent(out) :: geom
```

Comment:
For convenience we will eventually want grid (and mesh, xgrid, &  locstream) optional arguments here.  But not needed in this PR.